app/controllers/orders_controller.rb

```diff
@@ -23,16 +25,27 @@ def edit
 
   def show
     @order = Order.find(params[:id])
+    if (@order.user != current_user) && !current_admin
+      render file: '/public/404'
+    end
   end
 
   def update
+    @order = Order.find(params[:id])
+    @order.update(status: params[:status])
+    @order.update(date_time: DateTime.now)
+    redirect_to admin_dashboard_path
   end
 
   def destroy
   end
 
   private
     def order_params
-      params.require(:order).permit(:status, :total)
+      params.require(:order).permit(:status, :total, :purchaser_name, :address, :user_id)
+    end
+
+    def require_user
+      render file: '/public/404' unless current_user || current_admin
```

Comment:
Maybe I'm mistaken but if we have a current_admin then we should also have a current_user right? Do we need the || clause here?